sled-agent/src/illumos/zone.rs

```diff
@@ -24,50 +25,86 @@ pub const ZONE_PREFIX: &str = "oxz_";
 pub const PROPOLIS_ZONE_PREFIX: &str = "oxz_propolis-server_";
 
 #[derive(thiserror::Error, Debug)]
-pub enum Error {
-    // TODO: These could be grouped into an "operation" error with an enum
-    // variant, if we want...
-    #[error("Cannot halt zone: {0}")]
-    Halt(zone::ZoneError),
-
-    #[error("Cannot uninstall zone: {0}")]
-    Uninstall(zone::ZoneError),
-
-    #[error("Cannot delete zone: {0}")]
-    Delete(zone::ZoneError),
-
-    #[error("Cannot install zone: {0}")]
-    Install(zone::ZoneError),
-
-    #[error("Cannot configure zone: {0}")]
-    Configure(zone::ZoneError),
-
-    #[error("Cannot clone zone: {0}")]
-    Clone(zone::ZoneError),
+enum Error {
```

Comment:
Note: this isn't `pub` anymore.

For large enough error blocks, I've been finding "error-based-on-operation" to be a reasonable way to propagate errors and provide the precise amount of context, based on what was requested.